app/templates/scenario/scenario_step2.html

```diff
@@ -171,125 +168,144 @@ <h3>{{ group_names|get_item:group_name|title }}</h3>
-})
-        function clearGridModel() {
-            Swal.fire({
-                title: "Are you sure?",
-                text: "This will clear the whole grid model! This will not actually delete any asset from the scenario. You will need to save after clearing for the changes to actually take effect.",
-                icon: "warning",
-                showCancelButton: true,
-                confirmButtonText: "Yes, clear everything!",
-                cancelButtonText: "Cancel",
-            }).then((result) => result.value && editor.clearModuleSelected());
+    $(document).ready(function () {
+        disabled_assets = [];//["gas_dso", "h2_dso", "biogas_plant", "geothermal_conversion", "diesel_generator", "fuel_cell", "gas_boiler", "electrolyzer", "hess"];
+        disabled_assets.forEach(asset_name => {document.querySelector(`[data-node='${asset_name}']`).classList.add("greyed_out"); document.querySelector(`[data-node='${asset_name}']`).setAttribute("title", "This is disabled")});
+    });
+    function clearGridModel() {
+        Swal.fire({
+            title: "Are you sure?",
+            text: "This will clear the whole grid model! This will not actually delete any asset from the scenario. You will need to save after clearing for the changes to actually take effect.",
+            icon: "warning",
+            showCancelButton: true,
+            confirmButtonText: "Yes, clear everything!",
+            cancelButtonText: "Cancel",
+        }).then((result) => result.value && editor.clearModuleSelected());
+    }
+
+    function move_step(){
+        window.location.href = "{% url 'scenario_create_constraints' proj_id scen_id %}";
+    };
+
+    /*
+     * Export Topology to JSON and send to back-end.
+     * Used by the Save and Next buttons at page footer
+    */
+    function save_topology(move_step=false) {
+        // Data Pre-check
+        const editorData = editor.export().drawflow.Home.data;
+        const node_list = Object.values(editorData)
+        const node_names = node_list.map(obj => obj.data.name);
+        // warn in case of empty model
+        if (node_list.length == 0 && !confirm('No components in model. Continue?'))
+            return;
+        // Check if there are duplicate node names in the model
+        // and prevent user from saving the model if there are.
+        // Might also be handled in backend when saving names.
+        const duplicate_names = node_names.filter((name, i, a) => a.indexOf(name) !== i);
+        if (duplicate_names.length != 0) {
+            alert('There are nodes with duplicate names.\nRename nodes with names: ' + duplicate_names.toString());
+            return;
+        }
+        // Check for nodes to in nodesToDB
+        // A missing entry usually stems from missing values and blocks creation of drawflowData
+        const nodesNotInDB = node_list.filter(node => !nodesToDB.has('node-'+node.id));
```

Comment:
If the nodes' form save without error, then the component should simply be saved and only the one with missing parameters should appear on this list (ie a DSO should be able to be dragged and saved via the next or the save button). With the current implementation the user must click through all components and save them individually

Reply:
Nodes are now saved when added to editor (after drag-and-drop). It populates the modal (same as with double-click) and submits form. On success, node information is saved, allowing for model save or hitting "Next" without having to go into every single node. On error, modal opens with failed fields highlighted.